2.0.0

tjdraper opened the pull request.

tjdraper committed:
Move dev mode check to dependency injector

tjdraper committed:
Update test for dev mode change

tjdraper committed:
Remove session_start call. Start your own session if you want it

tjdraper committed:
Add short route collector variable $r

tjdraper committed:
Update composer.json config

tjdraper committed:
Move CSRF exempt segments config to composer.json key

tjdraper committed:
Add ability to disable action param middleware

tjdraper committed:
Add ability to disable CSRF middleware

tjdraper committed:
Update kernel invoke arguments to accept incoming middleware

tjdraper committed:
Update the Kernel to use the Container Interface

tjdraper committed:
Fix Kernel tests to be compatible with dependency on ContainerInterface

tjdraper committed:
Update the Action Param Router to use the Container Interface

tjdraper committed:
Implement emitter stack with conditional stream emitter

tjdraper committed:
Set version for upcoming release

tjdraper committed:
Update dependency on corbomite-config-collector to 1.1 so we don't need APP_BASE_PATH defined

tjdraper committed:
Update the readme for the upcoming release

kevinsmith commented on src/Kernel.php:
```diff
@@ -107,16 +162,18 @@ function (RouteCollector $routeCollector) use ($collector) {
+
+        $emitterStack->push($this->di->get(SapiEmitter::class));
+
+        $emitterStack->push($this->di->get(ConditionalSapiStreamEmitter::class));
```
Since `EmitterStack` and the emitters are all configured in this package's DI config, perhaps it would provide more flexibility to push the emitters onto the stack while setting up `EmitterStack` in the config rather than doing it here. That would allow an app that uses this package to override which emitters they want to use and in which order. (Apps can override or extend the DI config, right?)

tjdraper replied:
@kevinsmith You make a great point. Right now array_merge is used to pull together all the arrays of DI config. Unfortunately it works the other way because the base project's DI config is the first thing retrieved, then it merges in each of the others, which is simply a complete oversight on my part. It absolutely should work where the project has the final say on DI config. I'll have a stern conversation with myself.

The collector is a separate composer package because I use this methodology in a couple of packages. At some future date, I will implement a change to the order to array_merge the base project's config last, allowing it final say. But it shouldn't affect us for the moment.

As to doing the emitter stack in the DI config, that's a great point and I'll implement it that way.

tjdraper replied:
@kevinsmith https://github.com/buzzingpixel/corbomite-http/pull/1/commits/6b9e219a648680908586a6d8ae5ffa037bb9f08e

kevinsmith commented on readme.md:
```diff
@@ -8,47 +8,37 @@ Provides a light framework for responding to HTTP requests.
+If not in dev mode, and if you've provided a middleware to handle errors as described above, then your provided error middleware will be the first thing added to the middleware stack. In this way, you have a chance to render error pages in your app in production.
 
-## Error page class
+If you send a string class name, the Kernel will attempt to get the class from the [Corbomite Dependency Injector](https://github.com/buzzingpixel/corbomite-di) and fall back to `new`ing up the class.
```
I can't remember if I asked this the other day, but is there anything about this package that specifically requires https://github.com/buzzingpixel/corbomite-di or could it simply require PSR-11's interfaces? Up to you, of course, just curious if there's an opportunity for decoupling here.

tjdraper replied:
https://github.com/buzzingpixel/corbomite-http/pull/1/commits/be1fb3507123dd0cbde16a579d5f06795109de79

tjdraper replied:
(@kevinsmith  To be clear I left it as a composer dependency for now (one of the other packages relies on it at this time but I removed it as specifically mentioned or used)

tjdraper committed:
Completely remove reference and dependency on Corbomite DI specifically

tjdraper committed:
Configure the emitter stack in the DI config instead of directly in the Kernel

@kevinsmith must be thanked for pointing me to this better way. I am a dumb dumb for not doing it that way in the first place

tjdraper committed:
Add option to disable CSRF middleware only when in dev mode
